pkg/sql/internal_test.go

```diff
@@ -134,6 +139,155 @@ func TestSessionBoundInternalExecutor(t *testing.T) {
+	// Check that the application_name is set properly in the executor.
+	if rows, _, err := ie.Query(context.TODO(), "test-query", nil,
+		"SHOW application_name"); err != nil {
+		t.Fatal(err)
+	} else if len(rows) != 1 {
+		t.Fatalf("expected 1 row, got: %+v", rows)
+	} else if appName := string(*rows[0][0].(*tree.DString)); appName != expectedAppName {
+		t.Fatalf("unexpected app name: expected %q, got %q", expectedAppName, appName)
+	}
+
+	// Start a background query using the internal executor. We want to
+	// have this keep running until we cancel it below.
+	sem := make(chan struct{})
+	errChan := make(chan error)
+	go func() {
+		sem <- struct{}{}
+		_, _, err := ie.Query(context.TODO(),
+			"test-query",
+			nil, /* txn */
+			"SELECT pg_sleep(1337666)")
+		if err != nil {
+			errChan <- err
+			return
+		}
+	}()
+
+	<-sem
+
+	// We'll wait until the query appears in SHOW QUERIES.
+	// When it does, we capture the query ID.
+	var queryID string
+	testutils.SucceedsSoon(t, func() error {
+		rows, _, err := ie.Query(context.TODO(),
+			"find-query",
+			nil, /* txn */
+			// We need to assemble the magic string so that this SELECT
+			// does not find itself.
+			"SELECT query_id, application_name FROM [SHOW QUERIES] WHERE query LIKE '%337' || '666%'")
+		if err != nil {
+			return err
+		}
+		switch len(rows) {
+		case 0:
+			// The SucceedsSoon test may find this a couple of times before
+			// this succeeds.
+			return fmt.Errorf("query not started yet")
+		case 1:
+			appName := string(*rows[0][1].(*tree.DString))
+			if appName != expectedAppName {
+				return fmt.Errorf("unexpected app name: expected %q, got %q", expectedAppName, appName)
+			}
+
+			// Good app name, retrieve query ID for later cancellation.
+			queryID = string(*rows[0][0].(*tree.DString))
+			return nil
+		default:
+			return fmt.Errorf("unexpected results: %+v", rows)
+		}
+	})
+
+	// Check that the query shows up in the internal tables without error.
+	if rows, _, err := ie.Query(context.TODO(), "find-query", nil,
+		"SELECT application_name FROM crdb_internal.node_queries WHERE query LIKE '%337' || '666%'"); err != nil {
+		t.Fatal(err)
+	} else if len(rows) != 1 {
+		t.Fatalf("expected 1 query, got: %+v", rows)
+	} else if appName := string(*rows[0][0].(*tree.DString)); appName != expectedAppName {
+		t.Fatalf("unexpected app name: expected %q, got %q", expectedAppName, appName)
+	}
+
+	// We'll want to look at statistics below, and finish the test with
+	// no goroutine leakage. To achieve this, cancel the query. and
+	// drain the goroutine.
+	if _, err := ie.Exec(context.TODO(), "cancel-query", nil, "CANCEL QUERY $1", queryID); err != nil {
+		t.Fatal(err)
+	}
+	select {
+	case err := <-errChan:
+		if !isClientsideQueryCanceledErr(err) {
+			t.Fatal(err)
+		}
+	case <-time.After(time.Second * 5):
+		t.Fatal("no error received from query supposed to be canceled")
+	}
+
+	// TOOD(knz): remove this skip when we log internal queries in stats.
+	t.Skip("#32215")
```

Comment:
does this only skip the part of the test below it?

Reply:
yes it does (I checked the doc)